### PR TITLE
[DTensor] create default process group when absent

### PR DESCRIPTION
**Pull request by XilunWu**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #92069
* #91802
* #91801
* __->__ #91756

### CODE REVIEW

File: test/distributed/_tensor/test_device_mesh.py [hunk 1]
```diff
@@ -16,13 +19,32 @@
...
+        self.device_type = "cuda" if torch.cuda.is_available() else "cpu"
+        backend = "nccl" if self.device_type == "cuda" else "gloo"
+        # skip the test if not enough GPUs
+        if backend == "nccl" and torch.cuda.device_count() < self.world_size:
+            sys.exit(TEST_SKIPS[f"multi-gpu-{self.world_size}"].exit_code)
+        mesh_tensor = torch.arange(4).reshape(2, 2)
+        self.assertTrue(not is_initialized())
+        os.environ["MASTER_ADDR"] = "localhost"
+        os.environ["MASTER_PORT"] = "25364"
+        # mesh size can be smaller than world_size
```
Comment (wanchaol) [inline test/distributed/_tensor/test_device_mesh.py:40]: discussed offline, just do some simple assertion to ensure the num of ranks in mesh is the same as the world_size initially

Comment (XilunWu) [reply test/distributed/_tensor/test_device_mesh.py:40]: #91801

Comment (wanchaol) [reply test/distributed/_tensor/test_device_mesh.py:40]: nit: remove this comment?

**Review by wanchaol — APPROVED** [review]
looks good, one nit

**Pull request closed without merging**